### ISSUE TITLE
Make it clearer in --health what is controlled by the user and what is not 

### ISSUE THREAD

**Issue by A-Walrus** [helix-editor/helix#3287]
#### Describe your feature request
As can be seen by this issue #3177, it is not clear from the health output what issues _can be addressed by the user,_ for example installing the proper language server, and what is just the current state of support in helix, i.e. support for indent/textobject queries.
This distinction should be made clearer, both in `helix --health`, the table for all languages, and in `helix --health some-lang`, for a specific language.

**Comment by the-mikedavis** [comment]
I'm definitely open to suggestions on this but I don't have any ideas for conveying that info tersely. We could use more icons but then the icon meaning might become unintuitve/ambiguous. Surely that info can live in the wiki page for the health check ([here](https://github.com/helix-editor/helix/wiki/Healthcheck)) and then maybe the health check command should link out to the wiki page? IMO `hx --health` itself dumping out a wall of prose in the terminal seems like a bad experience

**Comment by nathan-at-least** [comment]
Hello, here to upvote the OP: I read over the docs for language support, and I know to use `--health <LANG>` and initially I just focused on getting LSP support working for key languages.

However, when I began editing YAML I found that the syntax highlighting seemed crude, and I thought I must not have syntax support, then I ran `hx --health yaml` and realized I didn't understand the output well, and I realized I had a slew of questions based on this output:

```
$ hx --health yaml
Configured language server: yaml-language-server
Binary for language server: 'yaml-language-server' not found in $PATH
Configured debug adapter: None
Highlight queries: ✓
Textobject queries: ✘
Indent queries: ✓
``` 

- Is the syntax highlighting in my case correctly working and just a bad theme/color palette for my use?
- Is syntax highlighting for YAML not working, and it's falling back on a crude generic delimiter-only coloring? (I believe vim/emacs do this.)
- Does LSP support affect syntax highlighting in any way?
- What do "highlight queries, TextObject queries, indent queries" from the health check actually mean?

I found this ticket and found the [wiki link](https://github.com/helix-editor/helix/wiki/Healthcheck) which was more helpful, so now I believe I do have syntax highlighting for YAML and it just doesn't fit my taste. (Specifically, values are the same color as keys regardless of any quoting used.)

I consider my uncertainty a UX snag that can be improved. Yes, a subset of users can figure everything out if they study a variety of sources enough, but wouldn't it work better if a larger set of users got the right information directly from simply using `hx`?

I would prioritize user-education sources in this order:

1. In-band feedback while using `hx`.
2. The output of `hx` commandline (such as `--health`)
3. The docs on https://docs.helix-editor.com/ (Tangent: I personally highly prioritize offline dev, so I build and open the mdbook locally, fwiw).
4. Github discussions and tickets.

I personally don't value github wikis. Why have yet a fifth location to look up information? Why isn't the wiki information in the main docs? I would not have discovered the wiki link if not for this ticket.

**Recommendation:**

Given the "sources of information priorities" above, here's what I would recommend:

- Rename "Highlight queries" in user-facing output to "syntax highlighting". (I couldn't tell if "highlight queries" meant something about some kind of editor query like searching or jump-to-definition, etc… and I believe "syntax highlighting" is more broadly understood.)
- Rename "TextObject queries" to something like "Language-aware navigation (jump-to-definition, etc…)". This one is admittedly harder to clearly describe.
- Rename "Indent queries" to "auto-formatting". I'm actually uncertain if `Indent queries` means both adjusting the cursor as I edit _and_ formatting on save or not, so the description should reflect whether that's either or both of these.
- Display _in-band_ while editing any file when certain features are unavailable on the status bar, e.g. "⚠ language features misconfigured" with some way to pop open online health in-band. Perhaps from the space menu.

Thoughts?

I realize this is pretty opinionated, especially the renaming. Maybe the last recommendation is the most likely to be desired by devs and users alike? That might be a fun PR if I had time to hack on `hx`. ;-)

**Comment by eduardorittner** [comment]
I agree with renaming "Highlight queries" to "syntax highlighting", "Language-aware navigation" for "TextObject queries" I'm not so sure about, but it's still clearer than not changing it in my opinion. With regards to "Indent queries", as far as I can tell they are only used for figuring out the indentation of a new line, e.g. when pressing 'o', 'O' in normal mode or 'Enter' in insert mode, so maybe something like "Smart indenting"?
